remotes/docker/authorizer.go

```diff
@@ -186,15 +186,17 @@ func (a *dockerAuthorizer) AddResponses(ctx context.Context, responses []*http.R
 				return err
 			}
 
-			if username != "" && secret != "" {
-				common := auth.TokenOptions{
-					Username: username,
-					Secret:   secret,
-				}
+			if username == "" || secret == "" {
+				return fmt.Errorf("%w: no basic auth credentials", ErrInvalidAuthorization)
+			}
 
-				a.handlers[host] = newAuthHandler(a.client, a.header, c.Scheme, common)
-				return nil
+			common := auth.TokenOptions{
+				Username: username,
+				Secret:   secret,
 			}
+
+			a.handlers[host] = newAuthHandler(a.client, a.header, c.Scheme, common)
```

Comment:
🙈 perhaps inline the option (but that's really a nit);

```suggestion
			a.handlers[host] = newAuthHandler(a.client, a.header, c.Scheme, auth.TokenOptions{
				Username: username,
				Secret:   secret,
			})
```